dotCMS/src/main/java/com/dotmarketing/factories/MultiTreeFactory.java

```diff
@@ -331,6 +331,35 @@ public static MultiTree getMultiTree(Identifier parent1, Identifier parent2, Ide
 			throw new DotRuntimeException(e.toString());
 		}
 	}
+
+	/**
+	 * Get a list of MultiTree for Contentlets using a specific Structure and specific Container
+	 * @param containerIdentifier
+	 * @param structureIdentifier
+	 * @return List of MultiTree
+	 */
+	public static List<MultiTree> getContainerStructureMultiTree(String containerIdentifier, String structureInode) {
+		try {
+			HibernateUtil dh = new HibernateUtil(MultiTree.class);
+			StringBuilder query = new StringBuilder();
+			query.append("FROM mt IN CLASS com.dotmarketing.beans.MultiTree ");
+			query.append("WHERE (mt.parent1 = ? or mt.parent2 = ?) ");
+			query.append("AND EXISTS (FROM c IN class com.dotmarketing.portlets.contentlet.business.Contentlet ");
+			query.append("  WHERE c.identifier = mt.child AND c.structureInode = ? )");
+
+			dh.setQuery(query.toString());
+			dh.setParam(containerIdentifier);
+			dh.setParam(containerIdentifier);
+			dh.setParam(structureInode);
+
+			return dh.list();
+
+		} catch (Exception e) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Catch a list of specific exception subtypes instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2221)